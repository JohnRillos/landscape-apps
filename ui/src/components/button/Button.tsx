import classNames from 'classnames';
import React, { ButtonHTMLAttributes } from 'react';

type ButtonVariant = 'primary' | 'secondary';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
}

const variantMap: Record<ButtonVariant, string> = {
  primary: 'text-white bg-gray-800 disabled:bg-gray-200 disabled:text-gray-400',
  secondary: '',
};

export default function Button({
  variant = 'primary',
  children,
  className,
  ...props
}: ButtonProps) {
  return (
    <button
      className={classNames('button', variantMap[variant], className)}
      {...props}
    >
      {children}
    </button>
  );
}
