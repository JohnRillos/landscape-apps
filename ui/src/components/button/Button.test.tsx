import { describe, it, expect } from 'vitest';
import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

describe('ChatInput', () => {
  it('renders as expected', () => {
    const { asFragment } = render(<Button>Hello!</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
