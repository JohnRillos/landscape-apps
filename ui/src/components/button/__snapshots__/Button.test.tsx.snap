// Vitest Snapshot v1

exports[`Button > renders as expected 1`] = `
<DocumentFragment>
  <button
    class="button text-white bg-gray-800 disabled:bg-gray-200 disabled:text-gray-400"
  >
    Hello!
  </button>
</DocumentFragment>
`;

exports[`ChatInput > renders as expected 1`] = `
<DocumentFragment>
  <button
    class="button text-white bg-gray-800 disabled:bg-gray-200 disabled:text-gray-400"
  >
    Hello!
  </button>
</DocumentFragment>
`;
